### PR TITLE
Squelch some dead code warnings.

### PR DESCRIPTION
**Pull request by rudkx**

### CODE REVIEW

**Comment by rudkx** [comment]
@swift-ci Please smoke test

File: lib/FrontendTool/FrontendTool.cpp [hunk 1]
```diff
@@ -393,7 +393,7 @@ LLVM_ATTRIBUTE_NOINLINE
 static void debugFailWithAssertion() {
   // This assertion should always fail, per the user's request, and should
   // not be converted to llvm_unreachable.
-  assert(0 && "This is an assertion!");
+  assert((0) && "This is an assertion!");
```
Comment (slavapestov) [inline lib/FrontendTool/FrontendTool.cpp]: llvm_unreachable is compiled in the same cases as asserts, so there's no point in not making this llvm_unreachable

File: lib/Parse/ParseType.cpp [hunk 2]
```diff
@@ -184,11 +184,11 @@ ParserResult<TypeRepr> Parser::parseTypeSimple(Diag<> MessageID,
     TypeSpecifier = VarDecl::Specifier::InOut;
   } else if (Tok.is(tok::identifier)) {
     if (Tok.getRawText().equals("__shared")) {
-      assert(false);
+      assert((false));
```
Comment (slavapestov) [inline lib/Parse/ParseType.cpp]: This and all the others should be llvm_unreachable

File: lib/FrontendTool/FrontendTool.cpp (hunk 1 above)
Comment (rudkx) [reply lib/FrontendTool/FrontendTool.cpp]: Presumably there's a reason why this specifically says it should not be llvm_unreachable.

File: lib/Parse/ParseType.cpp (hunk 2 above)
Comment (rudkx) [reply lib/Parse/ParseType.cpp]: Yes, true. Blindly trying to kill off warnings without paying enough attention. Thanks for noticing.

File: lib/FrontendTool/FrontendTool.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/FrontendTool/FrontendTool.cpp]: Yeah, a misunderstanding on the author's part :)

Comment (slavapestov) [reply lib/FrontendTool/FrontendTool.cpp]: Well, the original patch did this:

```
commit 9df53c5c71b5bb9252ee5d13cd349295a6d19602
Author: Connor Wakamo <cwakamo@apple.com>
Date:   Fri Feb 7 22:30:40 2014 +0000

    [frontend] Switch -debug-assert-* from llvm_unreachable() to assert(0).
    
    Swift SVN r13656
```

Comment (slavapestov) [reply lib/FrontendTool/FrontendTool.cpp]: The flag is only used in one test, test/Driver/assert.swift. If the test passes with it changed to an llvm_unreachable I would keep it that way.

Comment (rudkx) [reply lib/FrontendTool/FrontendTool.cpp]: llvm_unreachable actually has an effect in no-asserts build (which is why I never introduce it, and almost certainly why it was not used above).

It becomes __builtin_unreachable() which the optimizer takes to mean "party on!", which in practice for LLVM I believe means it deletes all code following this point until the next join in control flow.

File: lib/Parse/ParseType.cpp (hunk 2 above)
Comment (rintaro) [reply lib/Parse/ParseType.cpp]: Ah sorry I removed this part in https://github.com/apple/swift/pull/13173

**Comment by rudkx** [comment]
@swift-ci Please smoke test

**Pull request merged**